scripts/ccpp_prebuild.py

```diff
@@ -471,6 +471,21 @@ def compare_metadata(metadata_define, metadata_request):
                 var.convert_from(metadata_define[var_name][0].units)
             elif var.intent=='out':
                 var.convert_to(metadata_define[var_name][0].units)
+        # If the host model variable is allocated based on a condition, i.e. has an active attribute other
+        # than T (.true.), the scheme variable must be optional
+        if not metadata_define[var_name][0].active == 'T':
+            for var in metadata_request[var_name]:
+                if var.optional == 'F':
+                    logging.error("Conditionally allocated host-model variable {0} is not optional in {1}".format(
+                                  var_name, var.container))
+                    success = False
+        # TEMPORARY CHECK - IF THE VARIABLE IS ALWAYS ALLOCATED, THE SCHEME VARIABLE SHOULDN'T BE OPTIONAL
```

Comment:
Is this really temporary?

Reply:
Yes. This is only there because all models using ccpp_prebuild are in lockstep in terms of allocating or not allocating variables. I used this to check that there are no unnecessary `optional` variables in the physics.

But going forward, CESM may declare some variables as potentially allocated, whereas the UFS doesn't, or vice versa. We still need the schemes make this variable `optional`, but the UFS would have the `active` attribute.